pretyping/patternops.mli

```diff
@@ -42,7 +42,7 @@ val head_of_constr_reference : Evd.evar_map -> constr -> GlobRef.t
    a pattern; currently, no destructor (Cases, Fix, Cofix) and no
    existential variable are allowed in [c] *)
 
-val pattern_of_constr : Environ.env -> Evd.evar_map -> Constr.constr -> constr_pattern
+val pattern_of_constr : ?broken:bool -> Environ.env -> Evd.evar_map -> EConstr.constr -> constr_pattern
```

Comment:
Can we add some doc about what exactly is broken?

Reply:
Not sure I want to document this. No plugin should use the `broken` flag, and even without setting it it's dubious that it is legitimate to use `pattern_of_constr` at all. Most of the uses are working around a deficiency of the pattern implementation.